File: pytools/examples/golang/src/server/proto/ack_test_send_ok.go
package proto

import (
	"packet"
)

type AckTestSendOk struct {
	idU8                     uint8
	roleBase                 *MsgRoleBase
	idF32                    []float32
	idOpU8Flag               uint8
	idOpU8                   uint8
	opRoleBaseFlag           uint8
	opRoleBase               *MsgRoleBase
}

func (this *AckTestSendOk) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint8(this.idU8)
	pack.WriteBytes(this.roleBase.Encode())
	idF32Count := uint16(len(this.idF32))
	pack.WriteUint16(idF32Count)
	for i := uint16(0); i < idF32Count; i++ {
		pack.WriteFloat32(this.idF32[i])
	}
	pack.WriteUint8(this.idOpU8Flag)
	if this.idOpU8Flag == 1 {
		pack.WriteUint8(this.idOpU8)
	}
	pack.WriteUint8(this.opRoleBaseFlag)
	if this.opRoleBaseFlag == 1 {
		pack.WriteBytes(this.opRoleBase.Encode())
	}

	return pack.Encode(P_ACK_TEST_SEND_OK)
}

func (this *AckTestSendOk) SetIdU8(idU8 uint8) {
	this.idU8 = idU8
}

func (this *AckTestSendOk) SetRoleBase(roleBase *MsgRoleBase) {
	this.roleBase = roleBase
}

func (this *AckTestSendOk) SetIdF32(idF32 []float32) {
	this.idF32 = idF32
}

func (this *AckTestSendOk) SetIdOpU8(idOpU8 uint8) {
	this.idOpU8Flag = 1
	this.idOpU8 = idOpU8
}

func (this *AckTestSendOk) SetOpRoleBase(opRoleBase *MsgRoleBase) {
	this.opRoleBaseFlag = 1
	this.opRoleBase = opRoleBase
}

File: pytools/data/data.code.java.server/AckTestSendOk.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class AckTestSendOk
{
	private byte id_u8;
	private MsgRoleBase role_base;
	private List<Float> id_f32 = new ArrayList<Float>();
	private byte id_op_u8_flag;
	private byte id_op_u8;
	private byte op_role_base_flag;
	private MsgRoleBase op_role_base;


	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeByte(id_u8);
		packet.writeBytes(role_base.getBytes());
		int id_f32_count = id_f32.size();
		packet.writeShort((short)id_f32_count);
		for (int i = 0; i < id_f32_count; i++)
		{
			packet.writeFloat((float)id_f32.get(i));
		}
		packet.writeByte(id_op_u8_flag);
		if (id_op_u8_flag == 1)
		{
			packet.writeByte(id_op_u8);
		}
		packet.writeByte(op_role_base_flag);
		if (op_role_base_flag == 1)
		{
			packet.writeBytes(op_role_base.getBytes());
		}
		return packet.encode((short)Msg.P_ACK_TEST_SEND_OK);
	}



	public void setIdU8(byte id_u8)
	{
		this.id_u8 = id_u8;
	}

	public void setRoleBase(MsgRoleBase role_base)
	{
		this.role_base = role_base;
	}

	public void setIdF32(List<Float> id_f32)
	{
		this.id_f32 = id_f32;
	}

	public void setIdOpU8(byte id_op_u8)
	{
		this.id_op_u8_flag = (byte)1;
		this.id_op_u8 = id_op_u8;
	}

	public void setOpRoleBase(MsgRoleBase op_role_base)
	{
		this.op_role_base_flag = (byte)1;
		this.op_role_base = op_role_base;
	}

}
